docs/design-documents/hal/0005-pin-names-Arduino-Uno-standard.md

```diff
@@ -0,0 +1,144 @@
+
+### Arduino Uno (Rev3) connector Pins
+
+The following diagrams shows the Arduino Uno Rev3 standard for Mbed boards:
+
+![Static pinmap model](img/Arduino_Uno_Mbed.png)
+
+**Digital and Analog pin definition**
+
+The Arduino Uno (Rev3) form factor for Mbed boards should define D0-D15 & A0-A5 pins as part of the default standard. These pins should be defined in PinNames.h file within a PinName enum. The prefix `ARDUINO_UNO_` distinguishes these pins from pins defined for other custom or common connectors that may have similar pin names. 
+
+    // Arduino Uno (Rev3) connector pin connection naming  
+    // Px_xx relates to the processor pin connected to the Arduino Uno (Rev3) connector pin
+
+    ARDUINO_UNO_D0 = Px_xx,
+    ARDUINO_UNO_D1 = Px_xx,
+    ARDUINO_UNO_D2 = Px_xx,
+    ARDUINO_UNO_D3 = Px_xx,
+    ARDUINO_UNO_D4 = Px_xx,
+    ARDUINO_UNO_D5 = Px_xx,
+    ARDUINO_UNO_D6 = Px_xx,
+    ARDUINO_UNO_D7 = Px_xx,
+    ARDUINO_UNO_D8 = Px_xx,
+    ARDUINO_UNO_D9 = Px_xx,
+    ARDUINO_UNO_D10 = Px_xx,
+    ARDUINO_UNO_D11 = Px_xx,
+    ARDUINO_UNO_D12 = Px_xx,
+    ARDUINO_UNO_D13 = Px_xx,
+    ARDUINO_UNO_D14 = Px_xx,
+    ARDUINO_UNO_D15 = Px_xx,
+
+    ARDUINO_UNO_A0 = Px_xx,
+    ARDUINO_UNO_A1 = Px_xx,
+    ARDUINO_UNO_A2 = Px_xx,
+    ARDUINO_UNO_A3 = Px_xx,
+    ARDUINO_UNO_A4 = Px_xx,
+    ARDUINO_UNO_A5 = Px_xx,
+
+
+If the development board has the Arduino Uno connector in hardware, but does not comply with the Arduino Uno standard, whether it be with alternate functionality pins or no connected pins, the board should not be defined as Arduino Uno compliant and `ARDUINO_UNO` should not be added as a supported form factor in targets.json. Note this may result in a warning being generated at compile time to inform the user.
+
+The pins with alternate functions have to be referenced using MCU pin names (Px_xx) or pin name aliases in application code. 
+
+**I2C, SPI and UART definition**
+
+All I2C, SPI and UART pin name alias definitions for the Arduino Uno (Rev3) connector pins should be defined in ArduinoUnoAliases.h in the Mbed OS HAL (common to all Arduino Uno compliant targets) as follows:
+
+    #ifdef TARGET_FF_ARDUINO_UNO
+    // Arduino Uno I2C signals aliases
+    #define ARDUINO_UNO_I2C_SDA ARDUINO_UNO_D14
+    #define ARDUINO_UNO_I2C_SCL ARDUINO_UNO_D15
+
+    // Arduino Uno SPI signals aliases
+    #define ARDUINO_UNO_SPI_CS   ARDUINO_UNO_D10
+    #define ARDUINO_UNO_SPI_MOSI ARDUINO_UNO_D11
+    #define ARDUINO_UNO_SPI_MISO ARDUINO_UNO_D12
+    #define ARDUINO_UNO_SPI_SCK  ARDUINO_UNO_D13
+
+    // Arduino Uno UART signals aliases
+    #define ARDUINO_UNO_UART_TX  ARDUINO_UNO_D0
+    #define ARDUINO_UNO_UART_RX  ARDUINO_UNO_D1
+
+    #endif // TARGET_FF_ARDUINO_UNO
+
+**Using Arduino Uno pins from an application**
+
+> Pending to add details
+
```

Comment:
Explain details on how to use the Mbed library with a specific shield.
It would be good to explain how to customize.

Reply:
ex:
https://github.com/ARMmbed/mbed-os/blob/master/storage/blockdevice/COMPONENT_SD/mbed_lib.json#L5
https://github.com/ARMmbed/mbed-os/blob/master/connectivity/drivers/ble/FEATURE_BLE/COMPONENT_BlueNRG_MS/mbed_lib.json#L4
https://github.com/WIZnet-MbedEthernet/WIZnetInterface/blob/master/mbed_lib.json